etcdserver/server.go

```diff
@@ -689,7 +690,7 @@ func (s *EtcdServer) sync(timeout time.Duration) {
 // static clientURLs of the server.
 // The function keeps attempting to register until it succeeds,
 // or its server is stopped.
-func (s *EtcdServer) publish(retryInterval time.Duration) {
+func (s *EtcdServer) publish(retryInterval, timeout time.Duration) {
```

Comment:
It could mention that it always retry after the last one fails. adding parameter makes it complicated. will change.